app/models/QRCode.server.js

```diff
@@ -18,22 +18,16 @@ export async function getQRCodes(shop, graphql) {
     orderBy: { id: "desc" },
   });
 
-  if (!qrCodes.length) {
-    return qrCodes;
-  }
-
   return Promise.all(
-    qrCodes.map(async (qrCode) => supplementQRCode(qrCode, graphql))
+    qrCodes.map((qrCode) => supplementQRCode(qrCode, graphql))
   );
 }
 // [END get-qrcode]
 
 // [START get-qrcode-image]
 export async function getQRCodeImage(id) {
   const url = new URL(`/qrcodes/${id}/scan`, process.env.SHOPIFY_APP_URL);
-  const image = await qrcode.toBuffer(url.href);
-
-  return `data:image/jpeg;base64, ${image.toString("base64")}`;
```

Comment:
This is a funny one. `qrcode.toBuffer` returns PNG data, so this data URL has a the wrong content type. It doesn't cause issues because browsers ignore the image type and just content sniff.

However, `qrcode.toDataURL` already exists to do the work for us, and use the correct content-type.